lib/fluent/winsvc.rb

```diff
@@ -63,10 +63,13 @@ def service_main
     end
 
     def service_stop
-      set_event(@service_name)
-      if @pid > 0
-        Process.waitpid(@pid)
+      if @pid <= 0
+        set_event(@service_name)
+        return
       end
+
+      repeat_set_event_several_times_until_success(@service_name)
```

Comment:
It might exist better alternative name.

* wait_until_ready_for_supervisor_event_processing
* wait_until_ready_for_supervisor_event_handling

Any other good idea?

Reply:
Thanks! but it seems to me that the current name is a straightforward expression of what this method does.
What aspects do you feel are problematic?
Would it be easier to understand if the name reflects the purpose of the method?

>     * wait_until_ready_for_supervisor_event_processing
>     * wait_until_ready_for_supervisor_event_handling

For these names, I'm concerned about the following points.

* The main action of this method is `set_event`, not `wait`. (They don't tell us about `set_event`.)
* It is possible to give up waiting.

Reply:
How about this?: ddb51b814da90c6fe686a0522502ecf5b6167c07

Reply:
LGTM.